### PR TITLE
Specify keymap name

### LINKED ISSUE #239: --INSERT (eskk)-- makes more sense than --INSERT (lang)--
When you enable a keymap, its name is usually displayed in the insert mode indicator. E.g. after `:setlocal keymap=kana<CR>a` the indicator is `--INSERT (kana)--`. `eskk` makes it look like it enables some sort of a keymap. But since `'keymap'` option doesn't get changed, the indicator is `--INSERT (lang)--`. The value for the indicator is taken from (in the order of precedence) `b:keymap_name` variable, `'keymap'` option, or falls back to ["lang"][1]:

> When set to 1, the value of the "b:keymap_name" variable, the 'keymap' option or "<lang>" appears in the status line.

The idea behind `b:keymap_name` variable is to provide a shorter name if keymap filename is too [verbose][2]:

> The "b:keymap_name" can be set to a short name, which will be shown in the
status line.  The idea is that this takes less room than the value of
'keymap', which might be long to distinguish between different languages,
keyboards and encodings.

A good place to set `b:keymap_name` would be after the [following block][3]. What do you say? Should I create a PR?

[1]: https://neovim.io/doc/user/insert.html#i_CTRL-^
[2]: https://neovim.io/doc/user/mbyte.html#mbyte-keymap
[3]: https://github.com/tyru/eskk.vim/blob/1db7aceb2faf48b791bb4dbb31a34940e1967afe/autoload/eskk.vim#L1822-L1833

### PR DESCRIPTION
**Pull request by x-yuri**
Fixes #239.

### CODE REVIEW

File: autoload/eskk.vim [hunk 1]
```diff
@@ -1831,6 +1831,7 @@ function! eskk#enable() abort "{{{
         redrawstatus
         let ret = ''
     endif
+    let b:keymap_name = 'eskk'
```
Comment (Shougo) [inline autoload/eskk.vim:1834]: Please add the comment for it.

File: autoload/eskk.vim [hunk 2]
```diff
@@ -1831,6 +1831,11 @@ function! eskk#enable() abort "{{{
         redrawstatus
         let ret = ''
     endif
+
+    # Specify the "keymap" name
```
Comment (Shougo) [inline autoload/eskk.vim:1835]: The comment must be `"` instead of `#`.

Comment (x-yuri) [reply autoload/eskk.vim]: oh, indeed, forgot switch the "layout" (from ruby/bash/whatever) :)

**Comment by Shougo** [comment]
eskk.vim has internal mode.  You can get it by `eskk#get_mode()`.
I think it should be also displayed in modeline.

So I think

```vim
    let b:keymap_name = 'eskk-' . eskk#get_mode()
```

is better.  What do you think?

**Comment by Shougo** [comment]
Hm.  `b:keymap_name` should be changed in `eskk#set_mode()`.

**Comment by Shougo** [comment]
```diff
diff --git a/autoload/eskk.vim b/autoload/eskk.vim
index cb7ba2c..8ba8d98 100644
--- a/autoload/eskk.vim
+++ b/autoload/eskk.vim
@@ -1894,6 +1894,7 @@ function! eskk#set_mode(next_mode) abort "{{{
     endif
     " Change mode.
     let inst.mode = a:next_mode
+    let b:keymap_name = 'eskk-' . a:next_mode
     " Set cursor color.
     call eskk#set_cursor_color()
     " Clear preedit.
@@ -1906,8 +1907,9 @@ function! eskk#set_mode(next_mode) abort "{{{
                 \       st.init_phase
                 \       : g:eskk#preedit#PHASE_NORMAL)
                 \)
-    " For &statusline.
+    " For &statusline and &showmode.
     redrawstatus
+    echo ''
 endfunction "}}}
 function! eskk#get_mode() abort "{{{
     let inst = eskk#get_current_instance()
```

This is my fix.  Please test it.

**Comment by x-yuri** [comment]
I'm not sure what you know, and what you don't. So my explanations here may be needlessly detailed.

`eskk` doesn't provide a keymap in a usual sense. A keymap is essentially a mapping between sequences of English characters to sequences of non-English characters. E.g. the `kana` keymap (`/usr/share/vim/vim82/keymap/kana.vim` in my case):

```
:se keymap=kana<CR>aohayou
```

You can activate/deactivate a keymap by pressing <kbd>CTRL-^</kbd> (`i_CTRL-^`).

And idea to display the `eskk` mode in the insert mode indicator is a good one. But `vim` doesn't expect the keymap name to change e.g. after pressing <kbd>q</kbd>. Although if you press <kbd>CTRL-^</kbd> twice after that, you'll see the expected keymap name (`-- INSERT (eskk-kata) --`).

But rather than trying to work around it, I suggest using just `eskk` as the keymap name, and document displaying the mode in the status line.

Speaking of which, I've noticed that `eskk` provides a helper function to display the mode in the status line. Given the default status line:

```vim
set statusline=%<%f\ %h%m%r%=%-14.(%l,%c%V%)\ %P
```

One can add the `eskk` indicator this way:

```vim
set statusline=%<%f\ %h%m%r%{eskk#statusline()}%=%-14.(%l,%c%V%)\ %P
```

The function takes 2 parameters:

1. A format string for the case where `eskk` is enabled (default: `'[eskk:%s]'`). `%s` is replaced by the `eskk` mode.
2. What to display when `eskk` is disabled (default: `''`).

So to display `[%s]`/`[-]` instead one can do:

```vim
set statusline=%<%f\ %h%m%r%{eskk#statusline('[%s]','[-]')}%=%-14.(%l,%c%V%)\ %P
```

**Comment by Shougo** [comment]
OK.  It seems better.

**Comment by Shougo** [comment]
Merged.

**Pull request merged**